src/azure-cli/azure/cli/command_modules/appconfig/_validators.py

```diff
@@ -371,3 +373,28 @@ def validate_snapshot_import(namespace):
             raise InvalidArgumentValueError("--src-snapshot is only applicable when importing from a configuration store.")
         if any([namespace.src_key, namespace.src_label, namespace.skip_features]):
             raise MutuallyExclusiveArgumentError("'--src-snapshot' cannot be specified with '--src-key', '--src-label', or '--skip-features' arguments.")
+
+
+def validate_sku(namespace):
+    if namespace.sku.lower() == 'free':
+        if (namespace.enable_purge_protection or namespace.retention_days or namespace.replica_name or namespace.replica_location or namespace.no_replica):
+            logger.warning("Options '--enable-purge-protection', '--replica-name', '--replica-location' , '--no-replica' and '--retention-days' will be ignored when creating a free store.")
+            namespace.retention_days = None
+            namespace.enable_purge_protection = None
+            namespace.replica_name = None
+            namespace.replica_location = None
+            namespace.no_replica = None
+
```

Comment:
Does it make sense to directly return here after setting the namespace values to none when `sku` is free?

Reply:
Yeah makes sense. No need to continue validation. Updated